runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts

```diff
@@ -211,72 +186,59 @@ export default class ExcelActionHandler extends StatefulActionHandler {
         this.session.Worksheets(input.worksheet).Activate();
     }
 
-    async insertColumnsBefore(
-        input: ExcelInsertColumnsActionInput
-    ): Promise<void> {
+    async insertColumnsBefore(input: ExcelInsertColumnsActionInput): Promise<void> {
         const targetWorksheet = this.session.Worksheets(input?.worksheet ?? this.session.ActiveSheet.Name);
 
         try {
             const column = this.getColumnCoordinate(input.column);
             const amount = input.amount;
 
             targetWorksheet
-                .Range(
-                    targetWorksheet.Columns(column),
-                    targetWorksheet.Columns(column + amount - 1)
-                )
+                .Range(targetWorksheet.Columns(column), targetWorksheet.Columns(column + amount - 1))
                 .Insert();
         } catch (e) {
             throw new Error(ExcelErrorMessage.insertColumnsIncorrectInput(e));
         }
     }
 
-    async insertColumnsAfter(
-        input: ExcelInsertColumnsActionInput
-    ): Promise<void> {
+    async insertColumnsAfter(input: ExcelInsertColumnsActionInput): Promise<void> {
         const targetWorksheet = this.session.Worksheets(input?.worksheet ?? this.session.ActiveSheet.Name);
 
         try {
             const column = this.getColumnCoordinate(input.column);
             const amount = input.amount;
 
             targetWorksheet
-                .Range(
-                    targetWorksheet.Columns(column + 1),
-                    targetWorksheet.Columns(column + amount))
+                .Range(targetWorksheet.Columns(column + 1), targetWorksheet.Columns(column + amount))
                 .Insert();
         } catch (e) {
             throw new Error(ExcelErrorMessage.insertColumnsIncorrectInput(e));
         }
     }
 
-    async clearCells(
-        input: ExcelClearCellsActionInput
-    ): Promise<void> {
+    async clearCells(input: ExcelClearCellsActionInput): Promise<void> {
         try {
             const targetWorksheet = this.session.Worksheets(input?.worksheet ?? this.session.ActiveSheet.Name);
-            if (!Array.isArray(input.targetCells))
-                targetWorksheet
-                    .Range(input.targetCells)
-                    .Clear();
-            else for (const cellCoordinate of input.targetCells)
-                targetWorksheet
-                    .Range(cellCoordinate)
-                    .Clear();
+            if (!Array.isArray(input.targetCells)) targetWorksheet.Range(input.targetCells).Clear();
+            else for (const cellCoordinate of input.targetCells) targetWorksheet.Range(cellCoordinate).Clear();
         } catch (e) {
             throw new Error(ExcelErrorMessage.clearCellsIncorrectInput(e));
         }
     }
 
+    async deleteWorksheet(input: ExcelDeleteWorksheetActionInput): Promise<void> {
+
+        const targetWorksheet = this.session.Worksheets(input?.worksheet ?? this.session.ActiveSheet.Name);
```

Comment:
I think we may want to have the worksheet mandatory in this action - since it is deletion I think we should make user super aware about it and have him point what to delete. We will avoid situation where there were some changes made in previous actions, this worksheet stays active and the next action deletes this worksheet.

Reply:
This was intonational. There is an message that if worksheets name is not given active worksheet is deleted.

Reply:
Yes, there is a message, but I can imagine that when the user adds this action and see no errors before saving he may not read the tooltip. From process creation perspective I would prefer to be forced to provide this info as it will make me fully aware of what I am doing, but if that was intentional and well thought then ok - I rest my case. 😃